datasheets/client.py

```diff
@@ -111,23 +108,23 @@ def _retrieve_client_credentials(self):
             unexpanded_credential_path = os.environ.get('DATASHEETS_CREDENTIALS_PATH',
                                                         '~/.datasheets/client_credentials.json')
             credential_path = os.path.expanduser(unexpanded_credential_path)
-            store = oauth2client.file.Storage(credential_path)
-            # If the client credentials file does not yet exist then oauth2client will throw
-            # a warning that isn't useful; to avoid end-user confusion we suppress it
-            with warnings.catch_warnings():
-                warnings.simplefilter('ignore')
-                credentials = store.get()
+            try:
+                credentials = Credentials.from_authorized_user_file(credential_path)
+            except IOError:
+                credentials = self._fetch_new_client_credentials()
+                data = {
+                        "refresh_token": credentials.refresh_token,
+                        "client_id": credentials.client_id,
+                        "client_secret": credentials.client_secret
+                        }
+
+                with open(credential_path, "w+") as f:
+                    json.dump(data, f)
         else:
-            store = helpers._MockStorage()
-            credentials = None
-
-        if not credentials or credentials.invalid:
-            credentials = self._fetch_new_client_credentials(store)
-
-        self.email = credentials.id_token['email']
```

Comment:
In doing manual testing I realized that because we removed this line we now get a bunch of issues that sadly don't show up in tests because we mock self.email there. In client OAuth mode, we now have a broken `__repr__` and our error messages break for `_fetch_info_on_items()` and `_fetch_file_id()`. I looked around within the google-auth-python-client and google-auth-python-client-oauthlib libraries and I don't think there's actually a way to get the user's email address in non-service account mode, which blows and means that I think we have to update those broken items above to deal with this reality.

I think the easiest solution is just to assign self.email = "user's personal email address" here, as I _think_ that makes the error messages and the `__repr__` both comprehensible. It feels a bit sloppy, but I can't think of a better solution to be honest.

Note that several tests will need to be tweaked to account for the `self.email` change.

Reply:
@zcmarine Sorry for the long pause. School got crazy. I added the  self.email = "user's personal email address" Line. Which tests need fixing? I ran the test case after the change and they all passed.

Reply:
Pushed changes